AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m

```diff
@@ -93,77 +97,79 @@ - (void)enqueueItem:(id<MSLog>)item flags:(MSFlags)flags {
 
   // Internal ID to keep track of logs between modules.
   NSString *internalLogId = MS_UUID_STRING;
-
+  __weak typeof(self) weakSelf = self;
```

Comment:
> There is no dispatching so I am curious about why would we need a `weekSelf` here?

weakSelf is called further inside [autoreleasepool](https://github.com/microsoft/appcenter-sdk-apple/blob/948312efd6a850f49f204532aa53607fe50fab68/AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m#L106) block and [dispatch_async](https://github.com/microsoft/appcenter-sdk-apple/blob/948312efd6a850f49f204532aa53607fe50fab68/AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m#L126). I think we should leave this variable to avoid reference cycle.

Reply:
@ilyashumihin reference cycle is not a case when we do not store the reference to block.

Agree that this pattern is redundant in some cases, I'll re-visit this

Reply:
Revert unnecessary changes in this file - #2001